include/libopencm3/stm32/common/usart_common_fl_0.h

```diff
@@ -0,0 +1,302 @@
+#define USART_RQR_SBKRQ			(1 << 1)
+#define USART_RQR_ABRRQ			(1 << 0)
+
+/* USART_ISR Values ---------------------------------------------------------*/
+
+#define USART_ISR_REACK			(1 << 22)
+#define USART_ISR_TEACK			(1 << 21)
+#define USART_ISR_WUF			(1 << 20)
+#define USART_ISR_RWU			(1 << 19)
+#define USART_ISR_SBKF			(1 << 18)
+#define USART_ISR_CMF			(1 << 17)
+#define USART_ISR_BUSY			(1 << 16)
+#define USART_ISR_ABRF			(1 << 15)
+#define USART_ISR_ABRE			(1 << 14)
+#define USART_ISR_EOBF			(1 << 12)
+#define USART_ISR_RTOF			(1 << 11)
+#define USART_ISR_CTS			(1 << 10)
+#define USART_ISR_CTSIF			(1 << 9)
+#define USART_ISR_LBDF			(1 << 8)
+#define USART_ISR_TXE			(1 << 7)
+#define USART_ISR_TC			(1 << 6)
+#define USART_ISR_RXNE			(1 << 5)
+#define USART_ISR_IDLE			(1 << 4)
+#define USART_ISR_ORE			(1 << 3)
+#define USART_ISR_NF			(1 << 2)
+#define USART_ISR_FE			(1 << 1)
+#define USART_ISR_PE			(1 << 0)
+
+/* USART_ICR Values ---------------------------------------------------------*/
+
+#define USART_ICR_WUCF			(1 << 20)
+#define USART_ICR_CMCF			(1 << 17)
+#define USART_ICR_EOBCF			(1 << 12)
+#define USART_ICR_RTOCF			(1 << 11)
+#define USART_ICR_CTSCF			(1 << 9)
+#define USART_ICR_LBDCF			(1 << 8)
+#define USART_ICR_TCCF			(1 << 6)
+#define USART_ICR_IDLECF		(1 << 4)
+#define USART_ICR_ORECF			(1 << 3)
+#define USART_ICR_NCF			(1 << 2)
+#define USART_ICR_FECF			(1 << 1)
+#define USART_ICR_PECF			(1 << 0)
+
+/*****************************************************************************/
+/* API definitions                                                           */
+/*****************************************************************************/
+
+#define USART_PARITY			(USART_CR1_PCE | USART_CR1_PS)
+#define USART_PARITY_NONE		(0)
+#define USART_PARITY_EVEN		(USART_CR1_PCE)
+#define USART_PARITY_ODD		(USART_CR1_PCE | USART_CR1_PS)
+
+#define USART_MODE			(USART_CR1_TE | USART_CR1_RE)
+#define USART_MODE_NONE			(0)
+#define USART_MODE_RX			(USART_CR1_RE)
+#define USART_MODE_TX			(USART_CR1_TE)
+#define USART_MODE_TX_RX		(USART_CR1_TE | USART_CR1_RE)
+
+#define USART_FLOWCONTROL		(USART_CR3_RTSE | USART_CR3_CTSE)
+#define USART_FLOWCONTROL_NONE		(0)
+#define USART_FLOWCONTROL_RTS		(USART_CR3_RTSE)
+#define USART_FLOWCONTROL_CTS		(USART_CR3_CTSE)
+#define USART_FLOWCONTROL_RTS_CTS	(USART_CR3_RTSE | USART_CR3_CTSE)
+
+/*****************************************************************************/
+/* API Functions                                                             */
+/*****************************************************************************/
+
+BEGIN_DECLS
+
+void usart_set_baudrate(uint32_t usart, uint32_t baud);
+void usart_set_databits(uint32_t usart, uint32_t bits);
+void usart_set_stopbits(uint32_t usart, uint32_t stopbits);
+void usart_set_parity(uint32_t usart, uint32_t parity);
+void usart_set_mode(uint32_t usart, uint32_t mode);
+void usart_set_flow_control(uint32_t usart, uint32_t flowcontrol);
```

Comment:
That is correct, except that both are defined to be the same thing. The original created what were presumably easy to remember constants which have an identical name/function with the usart_common_all.c code. That way the call to usart_set_flow_control works the same on both the f0/l0 as well as the other chips from a function call level, but it uses a different register in the f0/l0 case. The `#define` is set up as the common constant.